### DISCUSSION TITLE
Possibility to import node positions/graph layout

### DISCUSSION THREAD

**Discussion by mathiasbockwoldt** [graphia-app/graphia#48] [Q&A] [answered]
It is quite possible I just didn't see this, but is it possible to load a graph with pre-defined node positions?

I have a graph with approx 250 000 nodes, so the layout algorithm is extremely slow on my machine. But fortunately, I have 3d coordinates associated with each single node. Is it possible to load these coordinates in Graphia and switch off the auto-layout algo?

**Comment by timangus** [comment, marked as answer]
(Been away for a while, apologies for the slow reply.) It is not no, sorry. We did consider using any layout information that is included in an imported file (such as graphml, gml etc.), but in many/most cases there is no standardised way of encoding layouts in such formats, so it would usually involve guess work. I'll perhaps revisit this at some point though. What format is it you were loading from?

**Reply by mathiasbockwoldt** [reply]
I work mostly with gml files. gml definitions are a bit patchy but the [original(?) definition](https://web.archive.org/web/20190303094704/http://www.fim.uni-passau.de:80/fileadmin/files/lehrstuhl/brandenburg/projekte/gml/gml-technical-report.pdf) suggested this format:
```
  node [
    id 0
    label "some label"
    graphics [
      fill "#ffff55" <-- the color
      x 4726082.0 <-- center coordinates
      y 7520207.0
      z 100769.5
    ]
  ]
```

This format is followed, e.g., by Cytoscape (not the third dimension, though).

**Comment by mathiasbockwoldt** [comment]
Coming back to Graphia, I now realized that it allows to export the layout. Would it be an easy option to allow this exact format to be imported again? There would be no guessing, no foreign formats. This would help me a lot already!

**Comment by timangus** [comment]
I added an option to load the exported layout back in again. You can try it here:
https://graphia.dev/Windows%20Installer/Graphia-dev-2682-ga66e7f54b-installer.exe
https://graphia.dev/macOS%20Disk%20Image/Graphia-dev-2682-ga66e7f54b.dmg
https://graphia.dev/Linux%20AppImage/Graphia-dev-2682-ga66e7f54b.tar.gz

**Reply by mathiasbockwoldt** [reply]
This is great, thank you! I will test this over the weekend.
